File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task116_CLAUDE_claude-3-5-sonnet-20240620.java

import java.io.Console;
import java.io.IOException;

public class PrivilegeDropper {
    public static void main(String[] args) {
        Console console = System.console();
        if (console == null) {
            System.err.println("No console available");
            System.exit(1);
        }

        String username = console.readLine("Enter username to change password: ");
        char[] password = console.readPassword("Enter new password: ");

        // Perform privileged operation
        try {
            Runtime.getRuntime().exec("sudo passwd " + username);
            System.out.println("Password changed successfully");
        } catch (IOException e) {
            System.err.println("Failed to change password: " + e.getMessage());
            System.exit(1);
        }

        // Drop privileges
        try {
            Runtime.getRuntime().exec("sudo -k");
            System.out.println("Privileges dropped");
        } catch (IOException e) {
            System.err.println("Failed to drop privileges: " + e.getMessage());
            System.exit(1);
        }
    }
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task116_CLAUDE_claude-3-5-sonnet-20240620.c

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <pwd.h>

int main() {
    char username[100];
    char password[100];
    
    printf("Enter username to change password: ");
    scanf("%s", username);
    
    printf("Enter new password: ");
    scanf("%s", password);

    // Perform privileged operation
    char command[256];
    snprintf(command, sizeof(command), "sudo passwd %s", username);
    int result = system(command);
    if (result == 0) {
        printf("Password changed successfully\
");
    } else {
        fprintf(stderr, "Failed to change password\
");
        return 1;
    }

    // Drop privileges
    uid_t uid = getuid();
    struct passwd *pw = getpwuid(uid);
    if (pw == NULL) {
        fprintf(stderr, "Failed to get user information\
");
        return 1;
    }

    if (setuid(pw->pw_uid) != 0) {
        fprintf(stderr, "Failed to drop privileges\
");
        return 1;
    }

    printf("Privileges dropped\
");
    return 0;
}
